internal/osutil/user.go

```diff
@@ -75,3 +79,40 @@ func UidGid(u *user.User) (sys.UserID, sys.GroupID, error) {
 
 	return sys.UserID(uid), sys.GroupID(gid), nil
 }
+
+// NormalizeUidGid returns the "normalized" UID and GID for the given IDs and
+// names. UID and GID take precedence over username and group. If user is
+// specified but not gid or group, the user's primary group ID is returned.
+func NormalizeUidGid(uid, gid *int, username, group string) (*int, *int, error) {
+	if uid == nil && username == "" && gid == nil && group == "" {
+		return nil, nil, nil
+	}
+	if uid == nil && username != "" {
+		u, err := userLookup(username)
+		if err != nil {
+			return nil, nil, err
+		}
+		n, _ := strconv.Atoi(u.Uid)
```

Comment:
from my earlier comment, this could become:
```
if username != "" {
  u, err := userLookup(username)
  if err != nil {
...
  }
  n, _ := strconv.Atoi(u.Uid)
  if uid != nil && *uid != n {
    return nil, nil, Error("user and user-id supplied, but they don't map to the same user %s => %d != %d")
  }

```

Thoughts?

Reply:
See above

Reply:
Per earlier rationale, let's please revisit this and fix it so we change both this and the API to conform to username precedence, or to raise an error until we have a plan we're all happy with in place.